### PR TITLE
Cleanup: Use variable types temperature_t and duration_t more consistently

### PR DESCRIPTION
**Pull request by sfuchs79**
### Describe the pull request:

- [ ] Bug fix
- [ ] Functional change
- [ ] New feature
- [x] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

This is a prerequisite for the numerical format feature...

Use struct temperature_t for temperatures in struct stats_t and
use get_temperature_string() when printing these temperatures for
statistics and HTML export.

Use duration_t for total_sac_time in struct stats_t

### CODE REVIEW

File: core/divelogexportlogic.cpp [hunk 1]
```diff
@@ -99,12 +99,15 @@ static void exportHTMLstatistics(const QString filename, struct htmlExportSettin
 			out << "\"AVG_SAC\":\"" << get_volume_string(stats_yearly[i].avg_sac) << "\",";
 			out << "\"MIN_SAC\":\"" << get_volume_string(stats_yearly[i].min_sac) << "\",";
 			out << "\"MAX_SAC\":\"" << get_volume_string(stats_yearly[i].max_sac) << "\",";
-			if ( stats_yearly[i].combined_count )
-				out << "\"AVG_TEMP\":\"" << QString::number(stats_yearly[i].combined_temp / stats_yearly[i].combined_count, 'f', 1) << "\",";
-			else
+			if ( stats_yearly[i].combined_count ) {
```
Comment (neolit123) [inline core/divelogexportlogic.cpp]: could be good opportunity to remove the extra spaces after `(` and before `)`.

File: core/divelogexportlogic.cpp [hunk 2]
```diff
@@ -99,12 +99,15 @@ static void exportHTMLstatistics(const QString filename, struct htmlExportSettin
...
+				temperature_t avg_temp;
+				avg_temp.mkelvin = stats_yearly[i].combined_temp.mkelvin / stats_yearly[i].combined_count;
+				out << "\"AVG_TEMP\":\"" << get_temperature_string(avg_temp) << "\",";
+			} else {
 				out << "\"AVG_TEMP\":\"0.0\",";
-			out << "\"MIN_TEMP\":\"" << ( stats_yearly[i].min_temp == 0 ? 0 : get_temp_units(stats_yearly[i].min_temp, NULL)) << "\",";
-			out << "\"MAX_TEMP\":\"" << ( stats_yearly[i].max_temp == 0 ? 0 : get_temp_units(stats_yearly[i].max_temp, NULL)) << "\",";
+			}
+			out << "\"MIN_TEMP\":\"" << ( stats_yearly[i].min_temp.mkelvin == 0 ? 0 : get_temperature_string(stats_yearly[i].min_temp)) << "\",";
+			out << "\"MAX_TEMP\":\"" << ( stats_yearly[i].max_temp.mkelvin == 0 ? 0 : get_temperature_string(stats_yearly[i].max_temp)) << "\",";
```
Comment (neolit123) [inline core/divelogexportlogic.cpp]: can we please remove the space `(`. if there are a lot of these in the file a separate whitespace cleanup patch can do it.

**Review by neolit123 — APPROVED** [review]
lgtm. only a couple of comments about whitespace.

i will leave it for someone else to double check. not sure if our unit tests cover the changed code.

**Comment by sfuchs79** [comment]
Whitespace cleanup done.

**Comment by dirkhh** [comment]
BTW, whenever you think to yourself "not sure if the unit tests cover this..." the best answer is to add a test that covers this for sure. Our test coverage continues to be way too small :-(

**Comment by dirkhh** [comment]
This looks fine to me, so let's merge.

**Pull request merged**